### DISCUSSION TITLE
关于四层负载均衡的方案

### DISCUSSION THREAD

**Discussion by dean916** [alibaba/xquic#86] [General]
在生产环境部署XQUIC，需要四层负载均衡（例如LVS）的兼容支持。
(例如：https://datatracker.ietf.org/doc/html/draft-duke-quic-load-balancers-06 ）
有推荐的做法吗？目前alibaba使用怎样的方案？

**Comment by Kulsk** [comment]
we follow the lb draft, the server will generate its own cid for lb, and the system defines the lb rule based on cid. whenever a quic packet is received, cid is used to dispatch this packet to its deatination.

**Reply by Lingtaonju** [reply]
Q1. Is there any details ?
Q2. Does the cid's format follow the quic lb draft ?
Q3. Does alibaba's Layer-4 LB has supported the QUIC-LB draft ?

**Reply by Kulsk** [reply]
1. as the server defined the length of it cid, when receiving a packet, server shall parse the cid first, as the draft specified, we used certain bytes in the cid as the key, to link this packet to a related process or RS.
2. yes.
3. I am not sure about the LB productions, LB is ever talked between several groups, maybe already supported.

**Reply by Kulsk** [reply]
server shall always choose its own cid following the lb draft when creating a connection.

**Reply by Lingtaonju** [reply]
Thanks for your reply.
For Question3, if the layer-4 lb has not supported the draft-quic-lb, how xquic supports  address-migration and multiple-path when cid changes in the same session? So, is there a  clear answer ?
